File: 01_Language/01_Functions/go/src/functions/pathinfo.go
package functions

import path2 "path"

func Pathinfo(path string) map[string]string {
	var m = map[string]string{}
	m["dirname"] = path2.Dir(path)
	m["basename"] = path2.Base(path)
	m["extension"] = path2.Ext(path)[1:]
	m["filename"] = path2.Base(path)[0:len(path2.Ext(path))]
	return m
}

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Pathinfo.java
package com.onevgo.functions;

import com.google.common.io.Files;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

public class Pathinfo {
    public static Map<String, String> pathinfo(String path) {
        HashMap<String, String> map = new HashMap<>();
        File file = new File(path);
        map.put("dirname", file.getParent());
        map.put("basename", file.getName());

        String extension = "";
        String filename = file.getName();
        int dotPos = file.getName().lastIndexOf('.');
        if (dotPos != -1) {
            extension = filename.substring(dotPos + 1);
            filename = filename.substring(0, dotPos);
        }
        map.put("extension", extension);
        map.put("filename", filename);
        return map;
    }

    public static void main(String[] args) {
        System.out.println(pathinfo("./test.txt"));
    }
}
